src/hint.c

```diff
@@ -787,6 +788,23 @@ void ReadWMName(ClientNode *np)
 
 }
 
+/** Read the machine for a client. */
+void ReadWMMachine(ClientNode *np)
+{
+   XTextProperty tprop;
+   char **tlist;
+   int tcount;
+   
```

Comment:
I think we want something along the lines of the following here:
```c
if(np->machineName) {
   Release(np->machineName);
   np->machineName = NULL;
}
```
I haven't looked carefully yet, but is machineName getting set to NULL when ClientNode is allocated and freed when deleted?

Reply:
You're right, i was missing that.